<a href="https://colab.research.google.com/github/cbarnes5/DATA-690-WANG/blob/main/assignment_09/assignment_09.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Assignment # 09 - Point Estimate and Interval Estimate (Confidence Interval)

A random survey of enrollment at **35** community colleges across the United States yielded the following figures: 

6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;

5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; 

17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; 

1,263; 7,285; 28,165; 5,080; 11,622

Perform point estimate and interval estimate with **95% confidence level** using **t-distribution**.

Since we don't know the population variance, we use t-distribiution instead of normal distribution.

In [2]:
# Import Python packages
import numpy as np
import math
import scipy.stats

### Step 0 - Data Preprocessing 

Process the raw data to make a list of integers. In order to calculate descriptive statistic, Python needs to work with a list of numbers.
### note:
Don't manually make the list by hand-typing the numbers. Write code to automate the data preparation.

In [3]:
# make each line of numbers a string object and then concatenate them together 
# The end result is one single string containing 35 numbers separated by ";"

data_1 = "6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;" 
data_2 = "5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; "
data_3 = "17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; "
data_4 = "1,263; 7,285; 28,165; 5,080; 11,622"
data = data_1 + data_2 + data_3 + data_4
data

'6,414; 1,550; 2,109; 9,350; 21,828; 4,300; 5,944; 5,722; 2,825; 2,044;5,481; 5,200; 5,853; 2,750; 10,012; 6,357; 27,000; 9,414; 7,681; 3,200; 17,500; 9,200; 7,380; 18,314; 6,557; 13,713; 17,768; 7,493; 2,771; 2,861; 1,263; 7,285; 28,165; 5,080; 11,622'

In [4]:
# Convert the single string to a list of strings using split() function
# Make sure to specify a delimter or separator
data=data.split(';')
data

['6,414',
 ' 1,550',
 ' 2,109',
 ' 9,350',
 ' 21,828',
 ' 4,300',
 ' 5,944',
 ' 5,722',
 ' 2,825',
 ' 2,044',
 '5,481',
 ' 5,200',
 ' 5,853',
 ' 2,750',
 ' 10,012',
 ' 6,357',
 ' 27,000',
 ' 9,414',
 ' 7,681',
 ' 3,200',
 ' 17,500',
 ' 9,200',
 ' 7,380',
 ' 18,314',
 ' 6,557',
 ' 13,713',
 ' 17,768',
 ' 7,493',
 ' 2,771',
 ' 2,861',
 ' 1,263',
 ' 7,285',
 ' 28,165',
 ' 5,080',
 ' 11,622']

Create a list of integers from the list of strings using List Comprehension or for loop. Make sure to remove the "," first and then convert the strings to integers.

In [5]:
# Using List Comprehension
data=[int(x.replace(',','')) for x in data]
data

[6414,
 1550,
 2109,
 9350,
 21828,
 4300,
 5944,
 5722,
 2825,
 2044,
 5481,
 5200,
 5853,
 2750,
 10012,
 6357,
 27000,
 9414,
 7681,
 3200,
 17500,
 9200,
 7380,
 18314,
 6557,
 13713,
 17768,
 7493,
 2771,
 2861,
 1263,
 7285,
 28165,
 5080,
 11622]

### Step 1 - Calculate and Display the Sample Size and Sample Mean

In [6]:
# Calculate and display the sample size
ss=len(data)
print('Sample size =', ss)

Sample size = 35


In [7]:
# Calculate and display the sample mean
mean=round(sum(data)/ss)
print('Sample mean =', mean)

Sample mean = 8629


The point estimate of the mean enrollment of US community colleges is **8629**.

### Step 2 - Calculate and Display the Sample Standard Deviation & Sample Standard Error

Sample Standard Deviation $S=\sqrt{\dfrac{1}{n-1}\sum\limits_{i=1}^n (X_i-\bar{X})^2}$

Sample Standard Error = $\dfrac{S}{\sqrt{n}}$

Note: The default **Delta Degree of Freedom (DDOF)** for Numpy's std function is 0 which is applicable to populate data. For sample data, we need to specify **ddof=1**. 

For the enrollment data, we round up the statistics to be the full integers (no decimal points).


In [8]:
# Calculate and display the sample standard deviation using Numpy's std function.
std=round(np.std(data, ddof=1))
print('Sample Standard Deviation =', std)

Sample Standard Deviation = 6944


In [9]:
# Calculate and display the sample standard error
se=round(std/math.sqrt(ss))
print('Sample Standard Error is', se)

Sample Standard Error is 1174


### Step 3 - Calculate t Critical Value using t-Distribution 

$\alpha$ = 1 - Confidence Level = 1 - 95% = 0.05

$\dfrac{\alpha}{2}$ = 0.025

n (sample size) = 35

df (degree of freedom) = n - 1 = 35 - 1 = 34

We will use Python scipy.stats t-distribution's PPF (Percentage Point Function) to calculate t critical value $t_{0.025,34}$.

In [10]:
# Calculate and display the t critical value using scipy.stats.t package ppf function
t=round(scipy.stats.t.ppf(q=1-.025,df=34), 2)
print('t critical value =', t)

t critical value = 2.03


### Step 4 - Calculate the Margin of Error

Margin of Error = t-Statistics * Sample Standard Error = $t_{\alpha/2,n-1}\left(\dfrac{s}{\sqrt{n}}\right)$

In [13]:
# Calculate and display the margin of error
moe=round(t*se)
print('Margin of Error =', moe)

Margin of Error = 2383


### Step 5 - Calculate Lower and Upper Limit of the Confidence Interval

Lower Limit = Sample Mean - Margin of Error

Upper Limit = Sample Mean + Margin of Error

In [14]:
# Calculate and display the lower limit
ll=mean-moe
print('Lower Limit =', ll)

Lower Limit = 6246


In [15]:
# Calculate and display the upper limit
ul=mean+moe
print('Upper Limit =', ul)

Upper Limit = 11012


### Step 6 - Now We have the 95% Confidence Interval
Confidence Interval ($\sigma$ unknown) = $\bar{x}\space\pm\space t_{\alpha/2}\left(\dfrac{S}{\sqrt{n}}\right)$ = Sample_Mean $\pm$ Margin of Error

In [16]:
# Print the 95% confidence interveral in the form (lower limit, upper limit)
string='The 95% Confidence Interval = ({}, {})'.format(ll, ul)
print(string)

The 95% Confidence Interval = (6246, 11012)


### The End